source/extensions/transport_sockets/tls/openssl_impl.cc

```diff
@@ -0,0 +1,173 @@
+#include "extensions/transport_sockets/tls/openssl_impl.h"
+
+#include <algorithm>
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "openssl/crypto.h"
+#include "openssl/hmac.h"
+#include "openssl/rand.h"
+#include "openssl/ssl.h"
+#include "openssl/x509v3.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Tls {
+
+int alpnSelectCallback(std::vector<uint8_t> parsed_alpn_protocols, const unsigned char** out,
+                       unsigned char* outlen, const unsigned char* in, unsigned int inlen) {
+  // Currently this uses the standard selection algorithm in priority order.
+  const uint8_t* alpn_data = &parsed_alpn_protocols[0];
+  size_t alpn_data_size = parsed_alpn_protocols.size();
+
+  if (SSL_select_next_proto(const_cast<unsigned char**>(out), outlen, alpn_data, alpn_data_size, in,
+                            inlen) != OPENSSL_NPN_NEGOTIATED) {
+    return SSL_TLSEXT_ERR_NOACK;
+  } else {
+    return SSL_TLSEXT_ERR_OK;
+  }
+}
+
+auto select_client_cert_cb = +[](SSL* ssl, X509** x509, EVP_PKEY** pkey) -> int {
+  std::cout << "!!!!!!!!!!!!!!!!!!!!!!!!!!!!! select_client_cert_cb \n";
```

Comment:
is this intended?

Reply:
This is from the original openssl work. I removed it when I made changes for this release (next PR).